src/components/visualization/bar-chart/BarChart.tsx

```diff
@@ -0,0 +1,120 @@
+import { ChartInput, DataRecord } from '@/types/visualization';
+import { AxisPair } from '@/types/configuration';
+import AxisSelector from './AxisSelector';
+import { Payload } from 'recharts/types/component/DefaultLegendContent';
+import { computeIfAbsent } from '@/utils/maputils';
+import { getColor } from '@/colors';
+import { useState } from 'react';
+
+let minValue: number;
+let maxValue: number;
+
+export default function BarChart({ chartInput }: { chartInput: ChartInput }) {
+  const axisPairs = chartInput.axisPairs as AxisPair[];
+  const [axesMap, setAxesMap] = useState(collectYAxes(axisPairs));
+  const [xAxis, setXAxis] = useState(axisPairs[0].xAxis);
+  console.log(chartInput.data);
+
+  function onLegendClick(e: Payload) {
+    setAxesMap(updateAxisMap(xAxis, e.dataKey?.toString(), axesMap));
+  }
+
+  function getBars() {
+    const yAxesMap = axesMap.get(xAxis) ?? new Map<string, boolean>();
+    return [...yAxesMap.entries()].map(([yAxis, visible], index) => (
+      // eslint-disable-next-line react/jsx-key
+      <Bar dataKey={yAxis} hide={!visible} fill={getColor(index)} />
+    ));
+  }
+
+  return (
+    <div>
+      <AxisSelector axesMap={axesMap} setAxis={setXAxis} />
+      <ResponsiveContainer debounce={200} aspect={chartInput.aspect}>
+        <BarChartRecharts
+          data={chartInput.data}
+          margin={{
+            top: 5,
+            right: 30,
+            left: 20,
+            bottom: 5,
+          }}
+        >
+          <CartesianGrid strokeDasharray="3 3" />
+          <XAxis dataKey={xAxis} label={xAxis} tick={false} />
+          <YAxis type="number" domain={getDomain(chartInput.data, axesMap, xAxis)} ticks={getTicks()} />
+          <ReferenceLine y={0} stroke="#000" />
+          <Tooltip />
+          <Legend onClick={onLegendClick} />
+          {getBars()}
+        </BarChartRecharts>
+      </ResponsiveContainer>
+    </div>
+  );
+}
+
+function getDomain(records: DataRecord, axesMap: Map<string, Map<string, boolean>>, xAxis: string) {
+  minValue = 0;
+  maxValue = 0;
+  const yAxesMap = axesMap.get(xAxis) ?? new Map<string, boolean>();
+  const visibleYAxes = new Set([...yAxesMap.entries()].filter(([_, visible]) => visible).map(([yAxis, _]) => yAxis));
+  records.forEach((record) => {
+    const valueArray = Object.entries(record)
+      .filter(([key, _]) => visibleYAxes.has(key))
+      .map((entry) => entry[1]);
+    for (const value of valueArray) {
+      const parsedValue = Number.parseFloat(value);
+      if (parsedValue > maxValue) {
+        maxValue = parsedValue;
+      } else if (parsedValue < minValue) {
+        minValue = parsedValue;
+      }
+    }
+  });
+  return [minValue, maxValue];
+}
+
+function collectYAxes(axisPairs: AxisPair[]) {
+  const axesMap = new Map<string, Map<string, boolean>>();
+  for (const axisPair of axisPairs) {
+    const yAxiForXAxis = computeIfAbsent(axesMap, axisPair.xAxis, new Map<string, boolean>()) as Map<string, boolean>;
```

Comment:
typo -> `yAxisForXAxis`

Reply:
Should actually be `yAxesForXAxis`, but thanks